server/tests/stt_ntb_ninjs_parse_test.py

```diff
@@ -77,19 +78,60 @@ def test_cv_lookup_function(self):
         assert _cv_lookup(cv_items, "") is None
 
     def test_fixture_structure(self):
-        """Test that the fixture file has the expected structure."""
-        # Test that the parsed item has the expected structure from the JSON fixture
+        """Test that the fixture file has the expected structure and mappings."""
+        # Basic structure from parsed item
         assert self.item["type"] == "text"
         assert (
             self.item["headline"]
             == "Food prices rise again as grocery giant expands into pharmacies"
         )
         assert self.item["byline"] == "Jecaterina Mantsinen"
         assert self.item["slugline"] == "test-imatrics"
-        # Test that subject data from fixture is present
-        assert "subject" in self.item or "anpa_category" in self.item
 
-        # Test that specific content from the JSON fixture is present
+        # --- Verify qcode mapping correctness (subject & anpa_category) ---
+        # subject/anpa_category must both exist and contain qcodes
+        subjects = self.item.get("subject") or []
+        anpa = self.item.get("anpa_category") or []
+
+        def _extract_qcodes(val):
+            if isinstance(val, dict):
+                val = [val]
+            if not isinstance(val, list):
+                return set()
+            return {
+                x.get("qcode") for x in val if isinstance(x, dict) and x.get("qcode")
+            }
+
+        subject_qcodes = _extract_qcodes(subjects)
+        anpa_qcodes = _extract_qcodes(anpa)
+
+        assert subject_qcodes, "subject should contain qcodes (non-empty)"
+        assert anpa_qcodes, "anpa_category should contain qcodes (non-empty)"
```

Comment:
I don't get it, so you get just qcodes but then those are not checked, could you do a test like:
```py
assert anpa_qcodes == {'a', 'b', 'c'}
assert subject_qcodes == ....
```